### ISSUE TITLE
Preference/Global FX/Volume Compensation - is it needed any more?

### ISSUE THREAD

**Issue by Gero5** [mpogue2/SquareDesk#1181]
I was running this at +3dB to compensate for pan/balance and EQ negative preamping see #944 
I cannot say since when for quite a while I noticed some overdriven sounds at certain music pieces but I never digged deeper into that.
with V1.0.9 I now did. my impression is - judging by ear only - that I get the same volume as from the QuickTime Player when this is set to 0dB. Can you pls check if that is true. the pan/Balance unit is not in the system any more, ok. what about the EQ? no negative preamp any more? +2dB still sound louder than Quicktime and distorted.

if this is the case than we do not need that anymore, do we? 
when set to 0 dB by default it won't harm. I suggest at least to modify the text.
<img width="249" alt="Bildschirmfoto 2024-09-11 um 22 37 21" src="https://github.com/user-attachments/assets/9d1f8041-84b5-4f0d-b468-6d4f1fa4dbb1">

**Comment by mpogue2** [comment]
Yep -- the "sin/cos" pan law we were using before is gone, and it does the simpler fade pan law that we discussed earlier.  So yes, we could change this to something like this:

"Compensates for the normal volume reduction introduced by the EQ controls (-2dB).  NOTE: Settings higher than 0dB can cause audible distortion/clipping in the audio signal, when EQ is set to boost frequencies."

Does that seem reasonable?

It's possible that the EQ would change the sound slightly even when at the nominal settings.
But, if we special-case that, then at the nominal setting there would be a 2dB boost, which might seem weird.  On the other hand, that might make it closer to what QuickTimePlayer does, when at the nominal positions.

**Comment by Gero5** [comment]
>volume reduction introduced by the EQ controls (-2dB)

are you sure that this is still the case? my impression is that there is no -2dB in the overall output volume when volume compensation is set to 0 dB.

I did another check with my high end headphones now. Reference is QuickTime.

volume:
- Preference/Global FX/Volume Compensation + 2dB - seems to be louder than QT
- Preference/Global FX/Volume Compensation  0dB - seems to be the same volume to me

Clearly hearable on headphones the music is distorted when played with the SquareDesk mp3 player. #1183

**Comment by mpogue2** [comment]
I'll post my results in more detail, once I've had a chance to try a bunch of different combinations.
So far, a quick test of a mono sine wave of amplitude 0.9 showed that the output (captured by Audio Hijack, which does a digital capture of what is sent to the speakers) has a max of 0.636 = about 3dB down from expected.

I'll try this with stereo, and some other combinations (like "Force Mono", which I almost always have turned on), before figuring out how best to improve this.

**Comment by mpogue2** [comment]
So far, with all controls at nominal:
- Stereo 1KHz tone, no Force Mono: 0.0dB difference between input and output
- Stereo 1KHz tone, WITH Force Mono: 0.0dB difference between input and output
- Mono 1KHz tone, no Force Mono: output 3dB lower than input
- Mono 1KHz tone, WITH Force Mono: output 3dB lower than input

**Comment by mpogue2** [comment]
So far, with all controls at nominal, using uncompressed WAV files @ 44.1KHz:

- Stereo 10Hz-20KHz chirp, no Force Mono: 0.0dB difference between input and output (all freq's)
- Stereo 10Hz-20KHz chirp, WITH Force Mono: 0.0dB difference between input and output  (all freq's)

So, there don't appear to be any frequency-dependent effects here.

**Comment by mpogue2** [comment]
EQ is already completely disabled, if the B/M/T controls are at zero.
BUT, SoundTouch (tempo/pitch) processing is still in the pipeline, even when the pitch and tempo are not being changed.

I have more conclusive results when looking at "Nora" (tempo/pitch at nominal), and matching up samples on the input side (WAV file generated from Audacity from MP3 file @ 24b/sample) vs samples on the output side (digitally captured by Audio Hijack @ 24b/sample):

WITHOUT SOUNDTOUCH (input and output samples match up almost exactly):
![AudacityvsSquareDesk_noST_24b](https://github.com/user-attachments/assets/7262a09e-da22-4dea-b5a8-fa09d256ffc2)

WITH SOUNDTOUCH ENABLED (some output samples are higher than input samples, as @gero5 suggested might be happening):
![AudacityvsSquareDesk_withST_24b](https://github.com/user-attachments/assets/59474591-0f92-4d7a-876c-8a0c690ece9c)

We can see that when SoundTouch is enabled, even when no tempo/pitch change is being done, there's still a small amount of sample increase for some samples near 1.0.

So, I would say that @gero5's thinking is correct, and it's likely caused by SoundTouch.  We could turn SoundTouch off when the tempo is unchanged (±0.5BPM, given that the tempo control is quantized to the nearest integer) and pitch is unchanged (0 semitones).

Before making this change, I'll send some sound samples to Gero for listening.

**Comment by mpogue2** [comment]
I've sent @gero5 three files to listen to:

- Nora MP3 file decompressed to WAV using Audacity
- Output from SquareDesk with EQ off, and SoundTouch DISABLED
- Output from SquareDesk with EQ off, and SoundTouch ENABLED (but tempo/pitch set to nominal)

**Comment by Gero5** [comment]
> volume reduction introduced by the EQ controls (-2dB)

to stay to the issue's topic here. Is -2dB preamp for the EQ still happening? I am just curious.

**Comment by mpogue2** [comment]
It looks like:
- there is a bug with MONO input files, where 3dB boost is needed
- with STEREO input files, there is no -2dB happening for EQ.

So, once the MONO bug is fixed, the text can change.  There may still be a need for temporary preamp, e.g. when the song is recorded at a low level, but volume normalization (which uses the PEAK of the song) fails to bring the volume up (because the entire track is low volume, but there's a single peak at say max volume of 1.0, preventing normalization).  

NOTE: I could work around that with a compressor, but I removed the previously-implemented compressor from SquareDesk, because nobody really used it.  Personally, I have some songs that are recorded low, with peaks preventing  normalization, so I could use the preamp to work around that.  It's a hack.

**Comment by Gero5** [comment]
> It looks like:

there is a bug with MONO input files, where 3dB boost is needed

Are you shure that it is a bug or is it just loudness equalization? If this is 3 dB less on each stereo channel compared to the mono signal then it plays the centred mono signal on same loudness  on a stereo loudspeaker system.
That is what I learned from our balance/pan discussion.

`If a mono production is played back on a stereo system (dual mono), it should be attenuated by 3 dB (= 3 LU) to sound as loud as the equivalent stereo production.`

[https://auphonic.com/blog/2020/06/09/loudness-normalization-mono-productions/#:~:text=If%20a%20mono%20production%20is,as%20the%20equivalent%20stereo%20production](https://auphonic.com/blog/2020/06/09/loudness-normalization-mono-productions/#:~:text=If%20a%20mono%20production%20is,as%20the%20equivalent%20stereo%20production.)

**Comment by mpogue2** [comment]
LOL -- yes, it's correct for MONO then...
